wal_test.go

```diff
@@ -14,427 +14,17 @@
-
-	testutil.Ok(t, w.cut())
-
-	// Cutting creates a new file.
-	testutil.Equals(t, 2, len(w.files))
-
-	testutil.Ok(t, w.write(WALEntrySeries, 1, []byte("Hello World!!")))
-
-	testutil.Ok(t, w.Close())
-
-	for _, of := range w.files {
-		f, err := os.Open(of.Name())
-		testutil.Ok(t, err)
-
-		// Verify header data.
-		metab := make([]byte, 8)
-		_, err = f.Read(metab)
-		testutil.Ok(t, err)
-		testutil.Equals(t, WALMagic, binary.BigEndian.Uint32(metab[:4]))
-		testutil.Equals(t, WALFormatDefault, metab[4])
-
-		// We cannot actually check for correct pre-allocation as it is
-		// optional per filesystem and handled transparently.
-		et, flag, b, err := newWALReader(nil, nil).entry(f)
-		testutil.Ok(t, err)
-		testutil.Equals(t, WALEntrySeries, et)
-		testutil.Equals(t, byte(walSeriesSimple), flag)
-		testutil.Equals(t, []byte("Hello World!!"), b)
-	}
-}
-
-func TestSegmentWAL_Truncate(t *testing.T) {
-	const (
-		numMetrics = 20000
-		batch      = 100
-	)
-	series, err := labels.ReadLabels(filepath.Join("testdata", "20kseries.json"), numMetrics)
-	testutil.Ok(t, err)
-
-	dir, err := ioutil.TempDir("", "test_wal_log_truncate")
-	testutil.Ok(t, err)
-	defer os.RemoveAll(dir)
-
-	w, err := OpenSegmentWAL(dir, nil, 0, nil)
-	testutil.Ok(t, err)
-	w.segmentSize = 10000
-
-	for i := 0; i < numMetrics; i += batch {
-		var rs []RefSeries
-
-		for j, s := range series[i : i+batch] {
-			rs = append(rs, RefSeries{Labels: s, Ref: uint64(i+j) + 1})
-		}
-		err := w.LogSeries(rs)
-		testutil.Ok(t, err)
-	}
-
-	// We mark the 2nd half of the files with a min timestamp that should discard
-	// them from the selection of compactable files.
-	for i, f := range w.files[len(w.files)/2:] {
-		f.maxTime = int64(1000 + i)
-	}
-	// All series in those files must be preserved regarding of the provided postings list.
-	boundarySeries := w.files[len(w.files)/2].minSeries
-
-	// We truncate while keeping every 2nd series.
-	keep := map[uint64]struct{}{}
-	for i := 1; i <= numMetrics; i += 2 {
-		keep[uint64(i)] = struct{}{}
-	}
-	keepf := func(id uint64) bool {
-		_, ok := keep[id]
-		return ok
-	}
-
-	err = w.Truncate(1000, keepf)
-	testutil.Ok(t, err)
-
-	var expected []RefSeries
-
-	for i := 1; i <= numMetrics; i++ {
-		if i%2 == 1 || uint64(i) >= boundarySeries {
-			expected = append(expected, RefSeries{Ref: uint64(i), Labels: series[i-1]})
-		}
-	}
-
-	// Call Truncate once again to see whether we can read the written file without
-	// creating a new WAL.
-	err = w.Truncate(1000, keepf)
-	testutil.Ok(t, err)
-	testutil.Ok(t, w.Close())
-
-	// The same again with a new WAL.
-	w, err = OpenSegmentWAL(dir, nil, 0, nil)
-	testutil.Ok(t, err)
-
-	var readSeries []RefSeries
-	r := w.Reader()
-
-	r.Read(func(s []RefSeries) {
-		readSeries = append(readSeries, s...)
-	}, nil, nil)
-
-	testutil.Equals(t, expected, readSeries)
-}
-
-// Symmetrical test of reading and writing to the WAL via its main interface.
-func TestSegmentWAL_Log_Restore(t *testing.T) {
```

Comment:
Why is this test deleted? Even though we don't log through the this WAL anymore, we'll have to support reading from it till 3.x atleast....

Reply:
re-added, one of these was failing that is why I removed them, but lucky enough it wasn't this one.